BullsAndCows/BullsAndCowsGame.cs

```diff
@@ -1,20 +1,58 @@
 ﻿using System;
+using System.Collections.Generic;
+using System.Linq;
 
 namespace BullsAndCows
 {
     public class BullsAndCowsGame
     {
-        private readonly SecretGenerator secretGenerator;
-        public BullsAndCowsGame(SecretGenerator secretGenerator)
+        private readonly ISecretGenerator secretGenerator;
+        private readonly string screit;
+        private int count;
+        private bool canContinue;
+
+        public BullsAndCowsGame(ISecretGenerator secretGenerator)
         {
             this.secretGenerator = secretGenerator;
+            this.screit = this.secretGenerator.GenerateSecret();
+            this.count = 0;
+            this.canContinue = true;
         }
 
-        public bool CanContinue => true;
-
         public string Guess(string guess)
         {
-            throw new NotImplementedException();
+            var secret = this.screit.ToCharArray();
+            var inputNumbers = guess.ToCharArray();
+            var inputNumberSet = inputNumbers.ToHashSet();
+            if (inputNumberSet.Count != inputNumbers.Length)
+            {
+                return "Wrong Input, input again";
+            }
+
+            if (inputNumbers.Length != 4)
+            {
+                return "Wrong Input, input again";
+            }
+
+            var match = secret.Select((t1, i) => inputNumbers.Where((t, j) => t == t1 && i == j).Count()).Sum();
+            var positionErrorMatch = secret.Select((t1, i) => inputNumbers.Where((t, j) => t == t1).Count()).Sum();
```

Comment:
Naming: 'positionErrorMatch' not only contains error matched digit, also contains correct matched digit
it's better to rename like:
match -> valueAndPositionMatchedCount
positionErrorMatch -> valueMatchedCount